rpcutils/rpcclient.go

```diff
@@ -261,6 +276,50 @@ func sideChainTips(allTips []dcrjson.GetChainTipsResult) (sideTips []dcrjson.Get
 	return
 }
 
+// SideChainFull gets all of the blocks in the side chain with the specified tip
+// block hash. The last block is not mainchain, but it's previous block is.
+func SideChainFull(client *rpcclient.Client, tipHash string) ([]string, error) {
+	// Do not assume specified tip hash is even side chain.
+	var sideChain []string
+
+	hash := tipHash
+	for {
+		header := GetBlockHeaderVerboseByString(client, hash)
+		if header == nil {
+			return nil, fmt.Errorf("GetBlockHeaderVerboseByString failed for block %s", hash)
+		}
+
+		// Main chain blocks have Confirmations != -1.
+		if header.Confirmations != -1 {
+			// The passed block is main chain, not a side chain tip.
+			if hash == tipHash {
+				return nil, fmt.Errorf("tip block is not on a side chain")
+			}
+			// This previous block is the main/side common ancestor.
+			break
+		}
+
+		// This was another side chain block.
+		sideChain = append(sideChain, hash)
+
+		// On to previous block
+		hash = header.PreviousHash
+	}
+
+	// Reverse side chain order so that last element is tip.
+	reverseStringSlice(sideChain)
+
+	return sideChain, nil
+}
+
+func reverseStringSlice(s []string) {
+	N := len(s)
+	for i := 0; i <= N/2-1; i++ {
```

Comment:
I think its less confusing when you use bracket where you do multiple different operations;
 N/2-1 -> (N/2)-1

Reply:
Personally I think this order of operations is pretty clear, but it's good to help readability if it doesn't affect code semantics.